### PR TITLE
MSEARCH-762 Local Instances are not exported with the cql file in Member tenant

### PR DESCRIPTION
**Pull request by viacheslavpoliakov**
### Purpose
Fix exporting for the cql file in Member tenant

### Approach
Add filter for not shared records of member tenant

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [ ] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
_List any Jira issues related to this pull request._
[MSEARCH-762](https://folio-org.atlassian.net/browse/MSEARCH-762)

### CODE REVIEW

File: src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java [hunk 1]
```diff
@@ -81,6 +81,22 @@ public SearchSourceBuilder convertForConsortia(String query, String resource, bo
...
+  /**
+   * Converts given CQL search query value to the elasticsearch {@link SearchSourceBuilder} object.
+   * Wraps base 'convert' and adds tenantId+not_shared filter in case of consortia mode
+   *
+   * @param query    cql query to parse
+   * @param resource resource name
+   * @param tenantId active affiliation member tenant name
+   * @return search source as {@link SearchSourceBuilder} object with query and sorting conditions
+   */
+  public SearchSourceBuilder convertNotSharedActiveAffiliation(String query, String resource, String tenantId) {
```
Comment (psmagin) [inline src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java:93]:
```suggestion
  public SearchSourceBuilder convertForConsortia(String query, String resource, String tenantId) {
```

File: src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java [hunk 2]
```diff
@@ -81,6 +81,22 @@ public SearchSourceBuilder convertForConsortia(String query, String resource, bo
...
+    var sourceBuilder = convert(query, resource);
+    var queryBuilder = consortiumSearchHelper.filterQueryForNotSharedActiveAffiliation(sourceBuilder.query(),
+      tenantId, resource);
```
Comment (psmagin) [inline src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java:96]:
```suggestion
    var queryBuilder = consortiumSearchHelper.filterQueryForActiveAffiliation(sourceBuilder.query(),
      resource, tenantId);
```

File: src/main/java/org/folio/search/service/ResourceIdService.java [hunk 3]
```diff
@@ -138,16 +138,23 @@ protected OutputStreamWriter createOutputStreamWriter(OutputStream outputStream)
...
 
     searchRepository.streamResourceIds(request, searchSource, idsConsumer);
   }
 
+  private SearchSourceBuilder getSearchSource(CqlResourceIdsRequest req, boolean notSharedActiveAffiliation) {
+    return notSharedActiveAffiliation
+      ? queryConverter.convertNotSharedActiveAffiliation(req.getQuery(), req.getResource(), req.getTenantId())
+      : queryConverter.convertForConsortia(req.getQuery(), req.getResource());
+  }
+
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/ResourceIdService.java:157]: I suggest to not do a check and always use `queryConverter.convertForConsortia(req.getQuery(), req.getResource())`

File: src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java [hunk 4]
```diff
@@ -40,6 +40,15 @@ public class ConsortiumSearchHelper {
...
 
+  public QueryBuilder filterQueryForNotSharedActiveAffiliation(QueryBuilder query, String tenantId, String resource) {
+    var centralTenantId = consortiumTenantService.getCentralTenant(tenantId).orElse("consortium");
+    var boolQuery = prepareBoolQueryForActiveAffiliation(query);
+
+    addActiveAffiliationClauses(boolQuery, tenantId, centralTenantId, resource, false);
+
+    return boolQuery;
+  }
+
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java:51]: `consortium` - is not always name of central tenant

Comment (psmagin) [inline src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java:51]:
```suggestion
        var centralTenantId = consortiumTenantService.getCentralTenant(tenantId);
    if (centralTenantId.isEmpty()) {
      return query;
    }
    return filterQueryForActiveAffiliation(query, contextTenantId, centralTenantId.get(), resource);
  }

```

File: src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java [hunk 5]
```diff
@@ -147,8 +156,8 @@ private BoolQueryBuilder prepareBoolQueryForActiveAffiliation(QueryBuilder query
   }
 
   private void addActiveAffiliationClauses(BoolQueryBuilder boolQuery, String contextTenantId,
-                                           String centralTenantId, String resource) {
-    var affiliationShouldClauses = getAffiliationShouldClauses(contextTenantId, centralTenantId, resource);
+                                           String centralTenantId, String resource, boolean isShared) {
+    var affiliationShouldClauses = getAffiliationShouldClauses(contextTenantId, centralTenantId, resource, isShared);
```
Comment (psmagin) [inline src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java:160]: I think you can keep previous version of the method

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java (hunk 1 above)
Comment (viacheslavpoliakov) [reply src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java]: done

File: src/main/java/org/folio/search/service/ResourceIdService.java (hunk 3 above)
Comment (viacheslavpoliakov) [reply src/main/java/org/folio/search/service/ResourceIdService.java]: done

File: src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java (hunk 4 above)
Comment (viacheslavpoliakov) [reply src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java]: fixed

Comment (viacheslavpoliakov) [reply src/main/java/org/folio/search/service/consortium/ConsortiumSearchHelper.java]: done

File: src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java (hunk 2 above)
Comment (viacheslavpoliakov) [reply src/main/java/org/folio/search/cql/CqlSearchQueryConverter.java]: done

File: NEWS.md [hunk 6]
```diff
@@ -28,6 +28,7 @@
 * Browse: Duplicate results in exact match with diacritics ([MSEARCH-751](https://folio-org.atlassian.net/browse/MSEARCH-751))
 * Classification browse: Fix instances count for Shared Facet ([MSEARCH-761](https://folio-org.atlassian.net/browse/MSEARCH-761))
 * Subjects/Contributors browse: Fix instances count for Shared Facet ([MSEARCH-782](https://folio-org.atlassian.net/browse/MSEARCH-782))
+* Local Instances are not exported with the cql file in Member tenant ([MSEARCH-762](https://folio-org.atlassian.net/browse/MSEARCH-762))
```
Comment (psmagin) [inline NEWS.md:31]:
```suggestion
* Search Resources IDs: Local instances are not searchable with requests from member tenants ([MSEARCH-762](https://folio-org.atlassian.net/browse/MSEARCH-762))
```

File: src/test/java/org/folio/search/service/ResourceIdServiceTest.java [hunk 7]
```diff
@@ -167,7 +167,7 @@ void streamResourceIds_positive_emptyCollectionProvided() throws IOException {
   @Test
   void streamResourceIdsInTextTextType_positive_emptyCollectionProvided() {
     mockSearchRepositoryCall(emptyList());
-    when(queryConverter.convertForConsortia(TEST_QUERY, RESOURCE_NAME)).thenReturn(searchSource());
+    when(queryConverter.convertForConsortia(TEST_QUERY, RESOURCE_NAME, TENANT_ID)).thenReturn(searchSource());
     when(properties.getScrollQuerySize()).thenReturn(QUERY_SIZE);
```
Comment (psmagin) [inline src/test/java/org/folio/search/service/ResourceIdServiceTest.java:171]: You changed existing tests to use the new method. However, an old method still exists. Please revert this test changes and right new tests for new method.

Comment (viacheslavpoliakov) [reply src/test/java/org/folio/search/service/ResourceIdServiceTest.java:171]: I've changed this method inside ResourceIdServiceTest.java and changed corresponding tests. 
Yes, the old method still exists, but inside this class it's never invoked.

File: NEWS.md (hunk 6 above)
Comment (viacheslavpoliakov) [reply NEWS.md:31]: fixed

**Review by viacheslavkol — APPROVED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**